### ISSUE TITLE
Hit assertion while running a job on a brand new eventhub partition or EndOfStream

### ISSUE THREAD

**Issue by SreeramGarlapati** [Azure/azure-event-hubs-spark#254]
Hit this assertion: when I ran the spark job against a brand new eventhub with NO data. 
This is a one-off scenario and I think customer will very rarely, if not never, run into this.

SPARK adapter hits this assertion while trying to build begin and end seqno’s out of the Partitions and since the EventHub is brand new – it builds zero results.

```
java.lang.AssertionError: assertion failed: translate: result size 0 does not equal partition count 32
at scala.Predef$.assert(Predef.scala:170) at org.apache.spark.eventhubs.client.EventHubsClient.translate(EventHubsClient.scala:248) at org.apache.spark.sql.eventhubs.EventHubsSource$$anonfun$initialPartitionSeqNos$1.apply(EventHubsSource.scala:103) at org.apache.spark.sql.eventhubs.EventHubsSource$$anonfun$initialPartitionSeqNos$1.apply(EventHubsSource.scala:101) at scala.Option.getOrElse(Option.scala:121) at org.apache.spark.sql.eventhubs.EventHubsSource.initialPartitionSeqNos$lzycompute(EventHubsSource.scala:101) at org.apache.spark.sql.eventhubs.EventHubsSource.initialPartitionSeqNos(EventHubsSource.scala:68) at org.apache.spark.sql.eventhubs.EventHubsSource.getOffset(EventHubsSource.scala:123) at org.apache.spark.sql.execution.streaming.MicroBatchExecution$$anonfun$5$$anonfun$apply$5.apply(MicroBatchExecution.scala:268) at org.apache.spark.sql.execution.streaming.MicroBatchExecution$$anonfun$5$$anonfun$apply$5.apply(MicroBatchExecution.scala:268) at org.apache.spark.sql.execution.streaming.ProgressReporter$class.reportTimeTaken(ProgressReporter.scala:306) at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:60) at org.apache.spark.sql.execution.streaming.MicroBatchExecution$$anonfun$5.apply(MicroBatchExecution.scala:267) at org.apache.spark.sql.execution.streaming.MicroBatchExecution$$anonfun$5.apply(MicroBatchExecution.scala:264) at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234) at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234) at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59) at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48) at scala.collection.TraversableLike$class.map(TraversableLike.scala:234) at scala.collection.AbstractTraversable.map(Traversable.scala:104) at org.apache.spark.sql.execution.streaming.MicroBatchExecution.org$apache$spark$sql$execution$streaming$MicroBatchExecution$$constructNextBatch(MicroBatchExecution.scala:264) at org.apache.spark.sql.execution.streaming.MicroBatchExecution.org$apache$spark$sql$execution$streaming$MicroBatchExecution$$populateStartOffsets(MicroBatchExecution.scala:237) at org.apache.spark.sql.execution.streaming.MicroBatchExecution$$anonfun$runActivatedStream$1$$anonfun$apply$mcZ$sp$1.apply$mcV$sp(MicroBatchExecution.scala:124) at org.apache.spark.sql.execution.streaming.MicroBatchExecution$$anonfun$runActivatedStream$1$$anonfun$apply$mcZ$sp$1.apply(MicroBatchExecution.scala:121) at org.apache.spark.sql.execution.streaming.MicroBatchExecution$$anonfun$runActivatedStream$1$$anonfun$apply$mcZ$sp$1.apply(MicroBatchExecution.scala:121) at org.apache.spark.sql.execution.streaming.ProgressReporter$class.reportTimeTaken(ProgressReporter.scala:306) at org.apache.spark.sql.execution.streaming.StreamExecution.reportTimeTaken(StreamExecution.scala:60) at org.apache.spark.sql.execution.streaming.MicroBatchExecution$$anonfun$runActivatedStream$1.apply$mcZ$sp(MicroBatchExecution.scala:121) at org.apache.spark.sql.execution.streaming.ProcessingTimeExecutor.execute(TriggerExecutor.scala:56) at org.apache.spark.sql.execution.streaming.MicroBatchExecution.runActivatedStream(MicroBatchExecution.scala:117) at org.apache.spark.sql.execution.streaming.StreamExecution.org$apache$spark$sql$execution$streaming$StreamExecution$$runStream(StreamExecution.scala:291) at org.apache.spark.sql.execution.streaming.StreamExecution$$anon$1.run(StreamExecution.scala:201)
```

**Comment by yorek** [comment]
Hi, I'm having the same problem...is the issue close 'cause it has been fixed?

**Comment by sabeegrewal** [comment]
@devang93 can you share your stack trace? I'm not able to repro this in 2.3.2!

**Comment by devang93** [comment]
Well actually.. I figured out the problem was with cluster. After restarting cluster and attaching the library 2.3.1 it works fine.

However, running into another issue while using 2.3.2 library.

I am reading some events data from azure eventhub using spark streaming over Azure Databricks.
When I use library 2.3.1 I am able to simple read the dataframe and display it over console using following code snippet:

val connectionString = "Valid EventHubs connection string."
val ehConf = EventHubsConf(connectionString)

val df = spark
  .readStream
  .format("eventhubs")
  .options(ehConf.toMap)
  .load()
  
val eventhubs = df.select($"body" cast "string")

eventhubs.writeStream.outputMode("append").format("console").start()

But when I switched to 2.3.2 library, I am not being able to see any records using the same code.
